synthetic_example.py

```diff
@@ -0,0 +1,141 @@
+import numpy as np
```

Comment:
Please remove this file `synthetic_example.py` from the PR. I don't think that the repo containing the package code is the right place for such a demonstration / example. We can transfer it into an example for the gallery (https://docs.doubleml.org/stable/examples/index.html). The source code for the example gallery can be found in the `doubleml-docs` repo, see https://github.com/DoubleML/doubleml-docs/tree/master/doc/examples.